### PR TITLE
Feat idempotent

### PR DESCRIPTION
**Pull request by YouriT**
When giving the right type of data to the `class-sanitizer`, it will throw errors because the underlying validator library only accepts strings as an input.

When building the sanitizer, you nevertheless accepted sometimes `any` inputs, therefore, to keep consistency, I implemented this fork to just return the input when it is already the appropriate type.

It has been done for: `toBoolean`, `toDate`, `toFloat` and `toInt`

The use case is with the class-validation, to be able to re-input the output and get validation.

I also updated `typescript` to `2.2.1` and got rid of `typings`, as there is no tests, I hope nothing broke!

@pleerock thanks for the great work on all your libraries :)

### CODE REVIEW

**Comment by YouriT** [comment]
P.S.: I'm using it through `npm link` for a couple of days, if you don't release I will release a `sw-class-sanitizer` and will `unpublish` when out :)

File: package.json [hunk 1]
```diff
@@ -23,9 +23,20 @@
     "typescript-sanitizer"
   ],
   "dependencies": {
+    "es6-promise": "^4.0.5",
+    "q": "^1.4.1",
```
Comment (pleerock) [inline package.json]: don't think we need to have "q" and es6-promise in dependencies.

Comment (pleerock) [reply package.json]: to use lib user either need to setup target to es6 either install es6-shim. We should not have these libs in dependency

File: package.json [hunk 2]
```diff
@@ -23,9 +23,20 @@
...
     "validator": "^5.1.0"
   },
   "devDependencies": {
+    "@types/assertion-error": "^1.0.30",
+    "@types/chai": "^3.4.35",
+    "@types/chai-as-promised": "^0.0.29",
+    "@types/es6-promise": "0.0.32",
```
Comment (pleerock) [inline package.json]: install es6-shim instead es6-promise

File: src/Sanitizer.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {SanitizeTypes} from "./SanitizeTypes";
 import {defaultMetadataStorage} from "./metadata/MetadataStorage";
 import {SanitizerInterface} from "./SanitizerInterface";
 import * as validatatorJs from "validator";
+import {Promise} from "es6-promise";
```
Comment (pleerock) [inline src/Sanitizer.ts]: don't do this (this will be fixed if `@types/es6-shim` will be installed)

File: src/Sanitizer.ts [hunk 4]
```diff
@@ -82,7 +83,7 @@ export class Sanitizer {
     /**
      * Canonicalize an email address.
      */
-    normalizeEmail(str: string, lowercase?: boolean): string {
+    normalizeEmail(str: string, lowercase?: boolean): string | false {
```
Comment (pleerock) [inline src/Sanitizer.ts:85]: why |false ?

**Comment by pleerock** [comment]
Hi, thank you for this PR, please review the comments

File: package.json (hunk 1 above)
Comment (YouriT) [reply package.json]: indeed, just forgot to remove it

File: src/Sanitizer.ts (hunk 4 above)
Comment (YouriT) [reply src/Sanitizer.ts:85]: The signature of `normalizeEmail` is as follow:
```ts
normalizeEmail(email: string, options?: NormalizeEmailOptions): string | false;
```

And if you look at [this](https://github.com/chriso/validator.js/blob/master/src/lib/normalizeEmail.js#L153) it indeeds returns false when it's not `isEmail`.

5.1 was different but still would return [false also](https://github.com/chriso/validator.js/blob/5.1.0/src/lib/normalizeEmail.js#L13)

File: package.json (hunk 2 above)
Comment (YouriT) [reply package.json]: Yup, done !

**Comment by YouriT** [comment]
All done @pleerock :)

File: package.json [hunk 5]
```diff
@@ -23,9 +23,19 @@
     "typescript-sanitizer"
   ],
   "dependencies": {
+    "es6-shim": "^0.35.3",
```
Comment (pleerock) [inline package.json]: as I told last time its user's choice to use es6-shim or target:es6. Shims never should be forced and user should decide if he will use it or not based on his environment. Thats why it should be removed from dependencies.

Comment (YouriT) [reply package.json]: indeed, my bad!

**Comment by YouriT** [comment]
here you are!

**Comment by YouriT** [comment]
@pleerock do you publish it too?

**Comment by pleerock** [comment]
Thank you. Released in `0.0.4`

**Comment by YouriT** [comment]
great! thanks :)

**Comment by YouriT** [comment]
mmh seems 0.0.4 is broken.
Check [this](https://registry.npmjs.org/class-sanitizer/-/class-sanitizer-0.0.4.tgz) there is nothing inside aside from `package.json`

**Pull request merged**